File: Cpp/1638_count_substrings_that_differ_by_one_character.cpp
// https://leetcode.cn/problems/count-substrings-that-differ-by-one-character/

class Solution {
public:
    int countSubstrings(string s, string t) {
        int res = 0;
        int m = s.size(), n = t.size();
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                int diff = 0;
                for (int k = 0; k < min(m-i, n-j); ++k) {
                    if (s[i+k] != t[j+k]) {
                        diff += 1;
                    }
                    if (diff == 1) {
                        res += 1;
                    }
                    else if (diff > 1) {
                        break;
                    }
                }
            }
        }
        return res;
    }
};

File: Python3/1638_count_substrings_that_differ_by_one_character.py
# https://leetcode.cn/problems/count-substrings-that-differ-by-one-character/

class Solution:
    def countSubstrings(self, s, t):
        """
        Inputs:
            s, t: str
        Outputs:
            res: int
        """
        res = 0
        m, n = len(s), len(t)
        for i in range(m):
            for j in range(n):
                diff = 0
                for k in range(min(m-i, n-j)):
                    if s[i+k] != t[j+k]:
                        diff += 1
                    if diff == 1:
                        res += 1
                    elif diff > 1:
                        break
        return res